applications/FreeSurfaceApplication/tests/generalTests.py

```diff
@@ -10,192 +12,62 @@ def GetFilePath(fileName):
-        fluid_solver.stabdt_convection_factor = 1.0e-02
-        fluid_solver.use_mass_correction = True
-        fluid_solver.tau2_factor = 1.0
-        fluid_solver.edge_detection_angle = 45.0
-        fluid_solver.assume_constant_pressure = False
-        fluid_solver.compute_porous_resistance_law = 0
+    def test_Reservoir2D(self):
+        self.__RunFreeSurfaceAnalysis("Reservoir2D", "reference_reservoir_2D")
 
-        fluid_solver.Initialize()
+    def test_Reservoir3D(self):
+        self.__RunFreeSurfaceAnalysis("Reservoir3D", "reference_reservoir_3D")
 
-        # settings to be changed
-        max_Dt = 1.0e-02
-        final_time = 5.0
-        safety_factor = self.safety_factor
+    def __RunFreeSurfaceAnalysis(self, test_name: str, reference_name: str):
+        with KratosUnittest.WorkFolderScope(test_name, __file__):
+            parameters_path = pathlib.Path("ProjectParameters.json")
+            model_part_path = pathlib.Path(test_name + ".mdpa")
+            with open(parameters_path, 'r') as file:
+                parameters = KratosMultiphysics.Parameters(file.read())
 
-        number_of_inital_steps = 10.0
-        initial_time_step = 1.0e-05
+            model = KratosMultiphysics.Model()
 
-        original_max_dt = max_Dt
+            analysis = FreeSurfaceAnalysis(model, parameters)
+            analysis.Run()
 
-        max_safety_factor = safety_factor
+            self.main_model_part = analysis._GetSolver().GetComputingModelPart()
+            self.checkResults(reference_name)
 
-        time = 0.0
-        step = 0
-
-        while(time < final_time):
-
-            if(step < number_of_inital_steps):
-                max_Dt = initial_time_step
-            else:
-                max_Dt = original_max_dt
-                # progressively increment the safety factor
-                # in the steps that follow a reduction of it
-                safety_factor = safety_factor * 1.2
-                if(safety_factor > max_safety_factor):
-                    safety_factor = max_safety_factor
-
-            Dt = fluid_solver.EstimateTimeStep(safety_factor, max_Dt)
-
-            time = time + Dt
-            self.main_model_part.CloneTimeStep(time)
-
-            if(step >= 3):
-                fluid_solver.Solve()
-
-                check_dt = fluid_solver.EstimateTimeStep(0.95, max_Dt)
-
-                if(check_dt < Dt):
-
-                    # we found a velocity too large! we need to reduce the time step
-                    fluid_solver.ReduceTimeStep(self.main_model_part, time)  # this is to set the database to the value at the beginning of the step
-
-                    safety_factor *= 3.00000e-01
-                    reduced_dt = fluid_solver.EstimateTimeStep(safety_factor, max_Dt)
+    def setUp(self):
+        self.check_tolerance = 1e-5
+        self.print_reference_values = False
 
-                    time = time - Dt + reduced_dt
+    def checkResults(self, reference_file_name: str):
```

Comment:
I know this was already like this, but FYI there are better / less manual / standard solutions for this in the Core